lib/util/propTypes.js

```diff
@@ -634,6 +634,8 @@ module.exports = function propTypesInstructions(context, components, utils) {
           // So we should construct an optional children prop
           this.shouldSpecifyOptionalChildrenProps = true;
 
+          this.shouldSpecifyClassNameProp = true;
```

Comment:
why do we need a class property for this?

Reply:
Since react generic type would have `className` too

Reply:
what i mean is, when is this not always true?

Reply:
So whenever the type is coming from react import reference, then we can have `className` too (similar to children).
When its not true is: type annotation, type literal, etc...

Reply:
`className` is just a prop, though, like any other. I'm a bit confused why it merits special treatment.

Reply:
`className` is a react specific prop for HTML elements. `class` is a vanilla HTML attribute.

Reply:
Sure, that's all true - but HTML elements don't use propTypes, so we're only talking about custom components here.

Reply:
@akulsr0 ok so, for custom components, `className` isn't special and is just like any other prop. For HTML elements, it is special and doesn't need a propType.

What are the next steps?

Reply:
I guess, should wrap this line inside an if condition for the later case.

Reply:
Added the condition, bumping up!